open_humans/views.py

```diff
@@ -372,33 +384,41 @@ def requesting_activities(self):
                     'joined': joined,
                     'members': project.authorized_members,
                 })
+        return activities
 
+    def requested_activities(self):
+        activities = {}
+        for source in self.project.request_sources_access:
+            proj = id_label_to_project(source)
+            if proj:
+                activities[source] = activity_from_data_request_project(
+                    proj, user=self.request.user)
         return activities
 
     def get_context_data(self, **kwargs):
         context = super(ActivityManagementView, self).get_context_data(
             **kwargs)
 
-        activities = personalize_activities_dict(self.request.user, only_active=False,
-                                                 only_approved=False)
         try:
-            self.activity = self.get_activity(activities)
+            self.project = DataRequestProject.objects.get(
+                slug=self.kwargs['source'])
         except KeyError:
             raise Http404
-
-        # MPB Aug 2017: This comprehension is slow, taking 2+ secs in local dev
-        pubfilecount_cachetag = 'pubfilecount-{}'.format(
-            self.activity['source_name'])
-        public_files = cache.get(pubfilecount_cachetag)
-        if not public_files:
-            public_files = len([
-                df for df in
-                DataFile.objects.filter(source=self.activity['source_name'])
-                .exclude(parent_project_data_file__completed=False)
-                .current().distinct('user') if df.is_public])
-        cache.set(pubfilecount_cachetag, public_files, timeout=TEN_MINUTES)
+        self.activity = activity_from_data_request_project(
+            self.project, user=self.request.user)
+
+        public_users = [
+            pda.user for pda in
+            PublicDataAccess.objects.filter(
+                data_source='direct-sharing-128').filter(
```

Comment:
this also should not be hard coded for the 23andMe upload. Rather should be `self.activity` I think